manticore/manticore.py

```diff
@@ -157,53 +169,27 @@ class Manticore(object):
     :type args: list[str]
     :ivar dict context: Global context for arbitrary data storage
     '''
-    def __init__(self, path_or_state, args=None, disasm='capstone'):
 
-        args = [] if args is None else args
+    def __init__(self, path_or_state, workspace_url=None, policy='random', **kwargs):
 
-        if isinstance(path_or_state, str):
-            assert os.path.isfile(path_or_state)
-            self._binary = path_or_state
-            self._initial_state_type = binary_type(self._binary)
+        if isinstance(workspace_url, str):
+            if ':' not in workspace_url:
+                ws_path = 'fs:' + workspace_url
+            else:
+                ws_path = workspace_url
         else:
-            assert isinstance(path_or_state, State), "Manticore must be intialized with either a state or a path to a binary"
-            self._binary = None
-            self._initial_state_type = 'MCORE'
-
-        self._check_disassembler_present(disasm)
-        self._disasm = disasm
-        # yan's comment
-        self._argv = args
-        self._env = {}
-        # Will be set to a temporary directory if not set before running start()
-        self._policy = 'random'
+            ws_path = None
```

Comment:
I think we should error here, rather than silently ignore and keep going

Reply:
I think ManticoreOutput will pick a random fresh workspace when None.

Reply:
everything will be fine, but a workspace_url that is not a string _is_ an error. i just got confused by this for a bit trying to test that api compat stuff because i was expecting some kind of error when i gave a list for workspace_url